gcloud/monitoring/test__dataframe.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    import pandas
+except ImportError:
+    HAVE_PANDAS = False
+else:
+    HAVE_PANDAS = True  # pragma: NO COVER
+
+import unittest2
+
+
+PROJECT = 'my-project'
+
+INSTANCE_NAMES = ['instance-1', 'instance-2']
+INSTANCE_ZONES = ['us-east1-a', 'us-east1-b']
+INSTANCE_IDS = ['1234567890123456789', '9876543210987654321']
+
+M = len(INSTANCE_NAMES)
+
+METRIC_TYPE = 'compute.googleapis.com/instance/cpu/utilization'
+METRIC_LABELS = list({'instance_name': INSTANCE_NAMES[i]}
+                     for i in range(M))
+
+RESOURCE_TYPE = 'gce_instance'
+RESOURCE_LABELS = list({'project_id': PROJECT,
+                        'zone': INSTANCE_ZONES[i],
+                        'instance_id': INSTANCE_IDS[i]}
+                       for i in range(M))
+
+METRIC_KIND = 'GAUGE'
+VALUE_TYPE = 'DOUBLE'
+VALUES = list(0.1 * i for i in range(M))
+
+TIMESTAMPS = [
+    '2016-04-06T22:05:00.042Z',
+    '2016-04-06T22:05:01.042Z',
+    '2016-04-06T22:05:02.042Z',
+]
+N = len(TIMESTAMPS)
+
+
+def parse_timestamps():  # pragma: NO COVER
+    import datetime
+    return [datetime.datetime.strptime(t, '%Y-%m-%dT%H:%M:%S.%fZ')
```

Comment:
No, I can't, because it adds a timezone.

I'm happy with having a self-contained one-liner (even if it is a little long), rather than calling out to a utility function in another file. Self-contained code is nicer to read.